app/src/main/java/io/homeassistant/companion/android/controls/MediaPlayerControl.kt

```diff
@@ -0,0 +1,98 @@
+package io.homeassistant.companion.android.controls
+
+import android.content.Context
+import android.os.Build
+import android.service.controls.Control
+import android.service.controls.DeviceTypes
+import android.service.controls.actions.BooleanAction
+import android.service.controls.actions.ControlAction
+import android.service.controls.actions.FloatAction
+import android.service.controls.templates.ControlButton
+import android.service.controls.templates.RangeTemplate
+import android.service.controls.templates.ToggleRangeTemplate
+import android.service.controls.templates.ToggleTemplate
+import androidx.annotation.RequiresApi
+import io.homeassistant.companion.android.common.R as commonR
+import io.homeassistant.companion.android.common.data.integration.Entity
+import io.homeassistant.companion.android.common.data.integration.IntegrationRepository
+import io.homeassistant.companion.android.common.data.integration.getVolumeLevel
+import io.homeassistant.companion.android.common.data.integration.getVolumeStep
+import io.homeassistant.companion.android.common.data.integration.supportsVolumeSet
+import java.math.BigDecimal
+import java.math.RoundingMode
+
+@RequiresApi(Build.VERSION_CODES.R)
+object MediaPlayerControl : HaControl {
+    override fun provideControlFeatures(
+        context: Context,
+        control: Control.StatefulBuilder,
+        entity: Entity<Map<String, Any>>,
+        info: HaControlInfo
+    ): Control.StatefulBuilder {
+        val isPlaying = entity.state == "playing"
+        if (entity.supportsVolumeSet()) {
+            val volumeLevel = entity.getVolumeLevel()
+            control.setControlTemplate(
+                ToggleRangeTemplate(
+                    entity.entityId,
+                    isPlaying,
+                    "",
+                    RangeTemplate(
+                        entity.entityId,
+                        volumeLevel?.min ?: 0f,
+                        volumeLevel?.max ?: 100f,
+                        volumeLevel?.value ?: 0f,
+                        entity.getVolumeStep(),
+                        "%.0f%%"
+                    )
+                )
+            )
+        } else {
+            control.setControlTemplate(
+                ToggleTemplate(
+                    entity.entityId,
+                    ControlButton(
+                        isPlaying,
+                        ""
+                    )
+                )
+            )
+        }
+        return control
+    }
+
+    override fun getDeviceType(entity: Entity<Map<String, Any>>): Int =
+        DeviceTypes.TYPE_TV
```

Comment:
This doesn't make a lot of sense for anything other than a tv like speakers, also because it sets the icon to a tv, but I don't see a good alternative 🤔

Reply:
yea was meaning to comment its the closest to anything media related except for `display`

Reply:
Checked Google Home: looks like they are showing a cast icon. Doesn't seem like it's built in so must be set manually (similar to how we override the icon if the user has set a different icon for the entity). I wonder if it is maybe worth always setting an icon here to better match the HA media player entity, or if I'm overreacting.

Reply:
> I wonder if it is maybe worth always setting an icon here to better match the HA media player entity, or if I'm overreacting.

my media players have different type of icons, we could go based off `device_class`? if we do that we have extra considerations as we are only overriding if icon is provided. Do we just first apply device_class then icon? Only do a special case for media player?

Reply:
I would only consider it for media player because it is hard to fit into the native UI (every manufacturer doing something different)

Reply:
ok will make sure to still respect the icon if it was overridden

Reply:
done and left a comment about the special case :)